backend/packages/Upgrade/src/api/services/CacheService.ts

```diff
@@ -1,43 +1,94 @@
 import { env } from './../../env';
 import { Service } from 'typedi';
 import cacheManager from 'cache-manager';
+import { redisStore, RedisCache } from 'cache-manager-ioredis-yet';
 import { CACHE_PREFIX } from 'upgrade_types';
 
-@Service()
-export class CacheService {
-  private memoryCache: cacheManager.Cache;
-  constructor() {
-    // read from the environment variable for initializing caching
-    let store: 'memory' | 'none';
+async function createMemoryCache() {
+  const store = "memory";
+  return cacheManager.caching({store, ttl: 900});
+}
+
+async function createNoCache() {
+  const store = "none";
+  return cacheManager.caching({store, ttl: 900});
+}
+
+async function createRedisCache() {
+  const redisCache: RedisCache = cacheManager.caching({store: await redisStore({
+          host: env.caching.redisHost,
+          port: env.caching.redisPort,
+          password: env.caching.redisPassword,
+        }), ttl: 900}) as any; // TODO: After updating the version of cache manager remove any
+
+  // Add event listener for connection to redis
+  const redisClient = redisCache.store.client;
+  redisClient.on('error', (err) => {
+        console.error('Redis error: ', err);
+  });
+
+  return redisCache;
+}
+
+async function createCache() {
+    let cache: cacheManager.Cache | RedisCache;
     if (env.caching.enabled) {
-      store = 'memory';
+      if (env.caching.redisHost && env.caching.redisPort) {
+        cache = await createRedisCache();
+      } else {
+        cache = await createMemoryCache();
+      }
     } else {
-      store = 'none';
+      cache = await createNoCache();
     }
-    this.memoryCache = cacheManager.caching({ store, max: 100, ttl: 900 });
+    return cache;
+}
+
+@Service()
+export class CacheService {
+  private cache: cacheManager.Cache | RedisCache;
+  constructor() {
+    this.init();
+  }
+
+  private async init() {
+    this.cache = await createCache()
   }
 
   public setCache<T>(id: string, value: T): Promise<T> {
-    return this.memoryCache ? this.memoryCache.set(id, value) : Promise.resolve(null);
+    return this.cache ? this.cache.set(id, value) : Promise.resolve(null);
   }
 
   public getCache<T>(id: string): Promise<T> {
-    return this.memoryCache ? this.memoryCache.get(id) : Promise.resolve(null);
+    return this.cache ? this.cache.get(id) : Promise.resolve(null);
   }
 
   public delCache(id: string): Promise<void> {
-    return this.memoryCache ? this.memoryCache.del(id) : Promise.resolve();
+    return this.cache ? this.cache.del(id) : Promise.resolve();
   }
 
   public async resetPrefixCache(prefix: string): Promise<void> {
-    const keys = this.memoryCache ? await this.memoryCache.store.keys() : [];
+    const keys = this.cache ? await this.cache.store.keys() : [];
     const filteredKeys = keys.filter((str) => str.startsWith(prefix));
-    this.memoryCache.store.del(filteredKeys);
+    this.cache.store.del(filteredKeys);
```

Comment:
This worked for me with an added test:
` if (filteredKeys.length > 0) {`
`    this.cache.store.del(filteredKeys);`
  `  }`

Seems like the `del()` function can't take an empty list.